src/writers.ts

```diff
@@ -32,10 +98,28 @@ export function createEvmWriters(indexerName: string) {
 
     payment.block = block.number;
     payment.type = metadata.type;
-    payment.beneficiary = payment.type === 'turbo' ? metadata.params.space : 'unknown';
+
+    if (payment.type === 'turbo') {
+      payment.space = metadata.params.space;
+    }
+
     payment.timestamp = block.timestamp;
 
     await payment.save();
+
+    // Try to get the space entity
+    let space = await Space.loadEntity(metadata.params.space, indexerName);
+    // If it doesn't exist, create it
+    if (!space) {
+      space = new Space(metadata.params.space, indexerName);
+    }
+
+    const expirationDate = computeExpiration(space, payment, metadata, block.timestamp);
```

Comment:
As `computeExpiration` is returning `new Date(0)` in case the sender has send less than one month amount, wouldn't this be set to timestamp 0, and moving back the expiration date for an existing space with a previously valid expiration date ?

Reply:
you are right. 
5b217bb6ef60f4a8cf8641f4c11c3ccd4e4b8978